### DISCUSSION TITLE
Video series or Boilerplate App

### DISCUSSION THREAD

**Discussion by arcanosam** [sqlpage/SQLPage#262] [General]
Hi,

I just think about the ideas below:

- a series of videos showing to build an complete application and deploying to production
   - complete in terms of all an business application today would have to be configured

- A Boilerplate Template with all basic and security features ready to custom to your needs

- a series of videos focusing on development to people who would like to help the creator

Sounds too crazy?

If I have to choose I would like to benefit first is videos series to learning

Thanks for your efforts to you all

Samuel

**Comment by lovasoa** [comment]
Hello  and welcome to SQLPage, @arcanosam !

  I do agree that learning ressources are currently a little bit lacking, especially for more advanced use-cases...
  
  With a young baby at baby at home, my time is limited, and I have to chose what I focus on carefully. Currently, it's the development of the application itself. We have made great progress lately, especially with the release of [`sqlpage.run_sql`](https://sql.ophir.dev/functions.sql?function=run_sql#function) today, but I will still focus on development at least until we have a better authentication story (easy secure session cookies, and OIDC).

  For deployment, I'd also like to spend time on [SQLPage cloud](https://sql.ophir.dev/your-first-sql-website/hosted.sql), which would have the additional advantage to crate a revenue stream to the project.

  I'm working on documentation and examples in parallel, and may eventually get to making videos, but I'd love to get help to get there faster, especially from people with a better command of English than me. @lukavia maybe ?

  If someone is trying to make their first contribution in that regard, here are a few topics that would be great to cover:
 - improving the main "get started" tutorial here: https://github.com/lovasoa/SQLpage/blob/main/examples/official-site/your-first-sql-website/tutorial.md
 - a small intro to forms: creating a form and inserting data, handling relationships, multi-step forms, file uploads
 - an intro to authentication: storing users, hashing passwords, handling sessions
 - analyzing data: creating menus, faceting, creating views, using the graph component
 - reading and writing json in SQL to create APIs, creating submenus and complex forms

**Reply by arcanosam** [reply]
I understand very well and I think it's great the focus you have chosen in development.

And sincerely, since I follow your project, I find awesome to see how much it evolved. You and the fellow colleagues who helped you really deserve a huge congratulations to delivery so much in such little time.

I saw in [another discussion](https://github.com/lovasoa/SQLpage/discussions/87) here about making a roadmap and from your answer I think you just set some milestones that already could be used:

> better authentication story (easy secure session cookies, and OIDC)

And all the topics you list:

> - improving the main "get started" tutorial here: https://github.com/lovasoa/SQLpage/blob/main/examples/official-site/your-first-sql-website/tutorial.md
> - a small intro to forms: creating a form and inserting data, handling relationships, multi-step forms, file uploads
> - an intro to authentication: storing users, hashing passwords, handling sessions
> - analyzing data: creating menus, faceting, creating views, using the graph component
> - reading and writing json in SQL to create APIs, creating submenus and complex forms


Even I know all efforts you all have been made, I would like to add one more suggestion: A simple page, at least for starting, about Development.

I suggest this because I remembered the [PgAdmin official page](https://www.pgadmin.org/development/) have one and I think it's an great way to get attention. Who knows? Could be attract more Rust developers to participate on your project.

Once more, thank you for your attention and I'm cheering about success of this project because I think it's one of fewer projects really trying to make web application development simpler.

**Comment by alexisrc1** [comment]
Hello @arcanosam,

I'm excited to share that we've launched the first series of video tutorials on how to build and deploy your SQLPage app. You can [check out the videos](https://www.youtube.com/playlist?list=PLTue_qIAHxAf9fEjBY2CN0N_5XOiffOk_), and [download all the files used in the tutorials](https://drive.google.com/drive/folders/13m3vUejJVKo9DrSU6yGLC3OUpPEzLcCD?usp=sharing).

I believe this is a good first step for anyone eager to learn how to build and deploy complete applications in just a few minutes.
Let me know what you think, more videos to come !

**Reply by axelmoe** [reply]
Fantastic, great job

**Comment by arcanosam** [comment]
Terrific news @alexisrc1 

Thanks for remember this discussion, I will check it soon!
